SportsHub/UnitTests/Controllers/TestUserController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Net;
+using System.Security.Claims;
+using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using SportsHub.Api.Controllers;
+using SportsHub.Api.DTOs;
+using SportsHub.AppService.Services;
+using SportsHub.Domain.Models;
+using UnitTests.MockData;
+using Xunit;
+
+namespace UnitTests.Controllers;
+
+public class TestUserController
+{
+    private readonly Mock<IUserService> _userService;
+    private readonly UserController _userController;
+    private readonly IMapper _mapper;
+
+    public TestUserController()
+    {
+        if (_mapper == null)
+        {
+            var mappingConfig = new MapperConfiguration(x => x.CreateMap<User, UserResponseDTO>());
+            IMapper mapper = mappingConfig.CreateMapper();
+            _mapper = mapper;
+        }
+
+        _userService = new Mock<IUserService>();
+        _userController = new UserController(_userService.Object, _mapper);
+    }
+
+    [Fact]
+    public async Task GetByUserNamelAsync_WithCorrectUser_ReturnsOkStatus()
```

Comment:
Typo.  `GetByUserNamelAsync_WithCorrectUser_ReturnsOkStatus` => `GetByUserNameAsync_WithCorrectUser_ReturnsOkStatus`

Reply:
I prefer to change the name to `GetByUserNameAsync_UserWithProvidedUsernameExists_ReturnsOkStatus`.